libtenzir/include/tenzir/healthmetrics_collector.hpp

```diff
@@ -0,0 +1,55 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "tenzir/fwd.hpp"
+
+#include "tenzir/actors.hpp"
+#include "tenzir/import_stream.hpp"
+#include "tenzir/plugin.hpp"
+
+#include <caf/typed_event_based_actor.hpp>
+
+namespace tenzir {
+
+/// Periodically run all registered health metrics
+class healthmetrics_collector_state {
+public:
+  // -- constants --------------------------------------------------------------
+
+  static inline constexpr auto name = "healthmetrics_collector";
+
+  // -- constructors, destructors, and assignment operators --------------------
+
+  healthmetrics_collector_state() = default;
+
+  using healthcheck = std::function<record()>;
+
+  // list of health checks to run
+  std::unordered_map<std::string, typename health_metrics_plugin::collector>
+    collectors;
+
+  // time between checks
+  caf::timespan collection_interval = std::chrono::seconds{30};
+
+  // output stream
+  std::unique_ptr<import_stream> importer = nullptr;
```

Comment:
FWIW: Feel free to replace the `import_stream` abstraction with something better if it gets in your way. If not, we can do it a later time.

Reply:
For this use case it seems pretty well suited so far, the only crutch is that the constructor may fail so we should run it delayed, but that's more of a general CAF interface issue.